test/components/views/settings/tabs/room/SecurityRoomSettingsTab-test.tsx

```diff
@@ -0,0 +1,402 @@
+import MatrixClientContext from "../../../../../../src/contexts/MatrixClientContext";
+import SettingsStore from "../../../../../../src/settings/SettingsStore";
+import { flushPromises, getMockClientWithEventEmitter, mockClientMethodsUser } from "../../../../../test-utils";
+import { filterBoolean } from "../../../../../../src/utils/arrays";
+
+describe("<SecurityRoomSettingsTab />", () => {
+    const userId = "@alice:server.org";
+    const client = getMockClientWithEventEmitter({
+        ...mockClientMethodsUser(userId),
+        getRoom: jest.fn(),
+        isRoomEncrypted: jest.fn(),
+        getLocalAliases: jest.fn().mockReturnValue([]),
+        sendStateEvent: jest.fn(),
+    });
+    const roomId = "!room:server.org";
+
+    const getComponent = (room: Room, closeSettingsFn = jest.fn()) =>
+        render(<SecurityRoomSettingsTab room={room} closeSettingsFn={closeSettingsFn} />, {
+            wrapper: ({ children }) => (
+                <MatrixClientContext.Provider value={client}>{children}</MatrixClientContext.Provider>
+            ),
+        });
+
+    const setRoomStateEvents = (
+        room: Room,
+        joinRule?: JoinRule,
+        guestAccess?: GuestAccess,
+        history?: HistoryVisibility,
+    ): void => {
+        const events = filterBoolean<MatrixEvent>([
+            new MatrixEvent({
+                type: EventType.RoomCreate,
+                content: { version: "test" },
+                sender: userId,
+                state_key: "",
+                room_id: room.roomId,
+            }),
+            guestAccess &&
+                new MatrixEvent({
+                    type: EventType.RoomGuestAccess,
+                    content: { guest_access: guestAccess },
+                    sender: userId,
+                    state_key: "",
+                    room_id: room.roomId,
+                }),
+            history &&
+                new MatrixEvent({
+                    type: EventType.RoomHistoryVisibility,
+                    content: { history_visibility: history },
+                    sender: userId,
+                    state_key: "",
+                    room_id: room.roomId,
+                }),
+            joinRule &&
+                new MatrixEvent({
+                    type: EventType.RoomJoinRules,
+                    content: { join_rule: joinRule },
+                    sender: userId,
+                    state_key: "",
+                    room_id: room.roomId,
+                }),
+        ]);
+
+        room.currentState.setStateEvents(events);
+    };
+
+    beforeEach(() => {
+        client.sendStateEvent.mockReset().mockResolvedValue({ event_id: "test" });
+        client.isRoomEncrypted.mockReturnValue(false);
+        jest.spyOn(SettingsStore, "getValue").mockRestore();
+    });
+
+    describe("join rule", () => {
+        it("warns when trying to make an encrypted room public", async () => {
+            const room = new Room(roomId, client, userId);
```

Comment:
This line is repeated quite a few times in these tests without any changes (I think). Perhaps it could be extracted to the enclosing `describe` as a variable? Or if we need a new room each run, maybe it could be created there then reassigned in the `beforeEach` hook.